packages/cli/src/lib/snapshot/snapshotReporter.ts

```diff
@@ -7,11 +7,43 @@ import {
 import {
   ReportViewerOperationName,
   ReportViewerResourceReportModel,
+  SnapshotReportStatus,
 } from './reportPreviewer/reportPreviewerDataModels';
 
 type ResourceEntries = [string, ResourceSnapshotsReportOperationModel];
 
+type NoopHandler = (this: void) => void;
+type SnapshotReporterHandler = (this: SnapshotReporter) => void | Promise<void>;
+
+type SnapshotReporterHandlers = FixableErrorHandlers &
+  UnfixableErrorHandlers &
+  SuccessfulReportHandler;
+
+type FixableErrorHandlers = {
+  [SnapshotReportStatus.MISSING_VAULT_ENTRIES]: /**
+   * TODO CDX-936: Define return type of the cb, & remove the SnapshotReporterHandler.
+   * Return type should be either a boolean if we do a 'all or nothing'
+   * handling (i.e. either we manage to do them all, or we do nothing)
+   * Return type should be a `Set<string>` if we do a 'best effort' strategy
+   * (i.e. if some entries are messed up, we ignore them and push the others)
+   */
```

Comment:
Too lazy to change the comment right now, will do tmrw:
 - [ ] Change comment @louis-bompart 
 
 But, @y-lakhdar , see `reportPreviewer` for an example of implementation (I think it makes more sense for the snapshot reporter to just be the orchestrator, and to leave the UI interactions to the reportPreviewer.